### PR TITLE
activate cql tests for cassandra 4 [tp-tests]

### LINKED ISSUE #3017: Dist tests failing
It looks like `dist-tests (8)` started to fail suddenly. I assume we use some external calls to download some dependencies / artifacts / data from external sources in that test and something was changes (maybe a URL or something else). Now all PR's and a `master` branch fails because of this change.

Here is the link to the failed test: https://github.com/JanusGraph/janusgraph/runs/6128877287

As seen from #3025 (https://github.com/JanusGraph/janusgraph/runs/6194674118?check_suite_focus=true) previously this test was working but after trying it again it started to fail.

For confirmed bugs, please report:
- Version: master

### Stack Trace

```
Error:  Failures: 
Error:    JanusGraphScriptIT.testGraphOfTheGodsGraphSONFull:33->JanusGraphAssemblyBaseIT.unzipAndRunExpect:113->JanusGraphAssemblyBaseIT.unzipAndRunExpect:84->JanusGraphAssemblyBaseIT.parseTemplateAndRunExpect:106->JanusGraphAssemblyBaseIT.expect:128->JanusGraphAssemblyBaseIT.command:171 expected: <0> but was: <1>
```

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #3017

* Upgrade Cassandra testing dist version to 4.0.6
* Enable full dist builds on java 11 tests
* Fix log collection during local testing
-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-dist/pom.xml [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
     <properties>
         <top.level.basedir>${project.basedir}/..</top.level.basedir>
+        <cassandra-dist.version>4.0.3</cassandra-dist.version>
```
Comment (FlorianHockmann) [inline janusgraph-dist/pom.xml:16]: Why did you add a specific `cassandra-dist.version` instead of upgrading the `cassandra.version`? This now means that we use `cassandra.version` (which is still `3.11.10`)  for tests, but `cassandra-dist.version` (`4.0.3`) in the zip distribution.
Shouldn't we always use the same version for our tests of the CQL backend as the one we include in the distribution?

File: janusgraph-dist/src/assembly/static/cassandra/conf/jvm-server.options [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2020 JanusGraph Authors
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/assembly/static/cassandra/conf/jvm-server.options:1]: The file has been renamed, but the `README.md` in `janusgraph-dist` still uses the old name.

File: janusgraph-dist/src/assembly/static/bin/janusgraph.sh [hunk 3]
```diff
@@ -83,7 +83,7 @@ wait_for_cassandra() {
     while [ $now_s -le $stop_s ]; do
         echo -n .
         # The \r\n deletion bit is necessary for Cygwin compatibility
-        statusbinary="`$BIN/../cassandra/bin/nodetool statusbinary 2>/dev/null | tr -d '\n\r'`"
+        statusbinary="`$BIN/../cassandra/bin/nodetool -Dcom.sun.jndi.rmiURLParsing=legacy statusbinary 2>/dev/null | tr -d '\n\r'`"
```
Comment (FlorianHockmann) [inline janusgraph-dist/src/assembly/static/bin/janusgraph.sh:86]: This disables new validations of URL parsing which seems to have been added for security reasons. Is it really necessary to disable this?

File: janusgraph-dist/pom.xml (hunk 1 above)
Comment (FlorianHockmann) [inline janusgraph-dist/pom.xml:16]: Apart from the two different versions: If we now ship with Cassandra 4 by default, then this should definitely be named in the `CHANGELOG.md`. It currently doesn't even mention that JanusGraph is compatible with Cassandra 4.

**Review by FlorianHockmann — CHANGES_REQUESTED** [review]

**Comment by porunov** [comment]
@farodin91 Thank you for the fixes! Do you think it's possible to split Cassandra version update (which I think is related to #2325) and dist tests fix (which is related to #3017) to 2 different PRs? I think we need to fix dist tests bug in `v0.6` branch. Otherwise we won't be able to release 0.6.x versions anymore.

File: janusgraph-dist/src/assembly/static/bin/janusgraph.sh (hunk 3 above)
Comment (farodin91) [reply janusgraph-dist/src/assembly/static/bin/janusgraph.sh:86]: It fixes the test.

**Comment by farodin91** [comment]
@FlorianHockmann  Would you like to review?

**Comment by FlorianHockmann** [comment]
> Would you like to review?

I just tried to review this again, but I'm still a bit confused: Is this now the PR to update Cassandra to version 4? In that case, this also fixes #2325 as @porunov noted, but then it should definitely be clearly documented in the changelog.

**Comment by farodin91** [comment]
cql hadoop can't be replaced directly. We have to switch to the spark read by datastax.

File: docs/changelog.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ All currently supported verions of JanusGraph are listed below.
 | JanusGraph | Storage Version | Cassandra | HBase | Bigtable | Elasticsearch | Solr | TinkerPop | Spark | Scala |
 | ----- | ---- | ---- | ---- | ---- | ---- | ---- | --- | ---- | ---- |
 | 0.6.z | 2 | 3.0.z, 3.11.z | 1.6.z, 2.2.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z, 1.14.z | 6.y, 7.y | 7.y, 8.y | 3.5.z | 3.0.z | 2.12.z |
-| 1.0.z | 2 | 3.0.z, 3.11.z | 1.6.z, 2.2.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z, 1.14.z | 6.y, 7.y | 8.y | 3.5.z | 3.0.z | 2.12.z |
+| 1.0.z | 2 | 3.11.z | 2.2.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z, 1.14.z | 6.y, 7.y | 8.y | 3.5.z | 3.0.z | 2.12.z |
```
Comment (FlorianHockmann) [inline docs/changelog.md:30]: Shouldn't 4.0.z also be added here as a Cassandra Version that 1.0.z is compatible with?

File: docs/changelog.md [hunk 5]
```diff
@@ -68,6 +68,11 @@ compile "org.janusgraph:janusgraph-core:1.0.0"
 * Apache TinkerPop 3.5.3
 * Java 8, 11
 
+**Installed version in JanusGraph with Embedded Cassandra and Elasticsearch**
```
Comment (FlorianHockmann) [inline docs/changelog.md:71]:
```suggestion
**Installed versions in the Pre-Packaged Distribution**
```

**Comment by farodin91** [comment]
@FlorianHockmann Would you like to review it again?

**Comment by farodin91** [comment]
@porunov Would you like to review this PR?

File: janusgraph-dist/src/assembly/static/cassandra/bin/cassandra [hunk 6]
```diff
@@ -82,27 +92,6 @@ elif [ -r "$CASSANDRA_INCLUDE" ]; then
...
-    done
-else
-    JAVA=java
-fi
-
-if [ -z $JAVA ] ; then
-    echo Unable to find java executable. Check JAVA_HOME and PATH environment variables. >&2
-    exit 1;
-fi
-
```
Comment (porunov) [inline janusgraph-dist/src/assembly/static/cassandra/bin/cassandra:105]: Why was this removed? I see `JAVA` variable is still used in this script.

**Review by porunov — COMMENTED** [review]
Thank you @farodin91 ! I have one comment below

File: janusgraph-dist/src/assembly/static/cassandra/bin/cassandra (hunk 6 above)
Comment (farodin91) [reply janusgraph-dist/src/assembly/static/cassandra/bin/cassandra:105]: This file is mostly https://github.com/apache/cassandra/blob/trunk/bin/cassandra with some lines changed.

Comment (porunov) [reply janusgraph-dist/src/assembly/static/cassandra/bin/cassandra:105]: I see it was removed here: https://github.com/apache/cassandra/commit/6ba2fb9395226491872b41312d978a169f36fcdb#diff-ca56a2b54ac847b9ff3908aac256aaa7f160680271725927a4b1917e64206011

Not sure why, but I guess that's fine

**Review by porunov — APPROVED** [review]
LGTM. Thank you @farodin91 !

File: janusgraph-dist/pom.xml (hunk 1 above)
Comment (farodin91) [reply janusgraph-dist/pom.xml]: I will cleanup those variables, if i have removed the `cassandra-all` dependency.

**Comment by farodin91** [comment]
@FlorianHockmann I've fixed both comments.

**Review by FlorianHockmann — APPROVED** [review]
